#167783727 View User Profile

JuwonAbiola opened the pull request:
#### What does this PR do?
- It enables a user to view other users profile

#### Description of Task proposed in this pull request?
- Create a controller method that views users profile
#### How should this be manually tested (Quality Assurance)?
- on your postman make a GET request to /profiles/:id

#### What are the relevant pivotal tracker stories?
- [167783727](https://www.pivotaltracker.com/story/show/167783727)

#### Any background context you want to add (Operations Impact)?
- N/A

#### What I have learned working on this feature:
- I learnt how to stub using sandbox

#### Screenshots: 
![image](https://user-images.githubusercontent.com/42170732/62773292-b622b000-ba99-11e9-9f9b-bd1615682ec4.png)

chialuka commented on server/docs/authors-haven-api.yml:
```diff
@@ -364,6 +364,32 @@ paths:
           description: Success. Email verification was successful
         500:
           description: Internal server errorcomponents     
+  /api/v1/users/profile:
```
It appears this route is incorrect. The `routes/profiles.js` page requires a param to work. This returns a 404

chialuka commented on server/docs/authors-haven-api.yml:
```diff
@@ -364,6 +364,32 @@ paths:
+          description: user does not exist
+        500:
+          description: Internal server error
+  /api/v1/users/profile/2:
```
Is the profiles route mounted on `/users/profiles`? From your routes, it is accessed via `/profiles`. The route in this documentation does not exist

JuwonAbiola replied:
Thanks for the feedback but this PR is still in progress

dinobi commented on server/controllers/Profiles.js:
```diff
@@ -19,7 +19,10 @@ class Profiles {
     try {
       const {
-        body, user, file, params
+        body,
```
why breaking into new lines?

dinobi commented on server/controllers/Profiles.js:
```diff
@@ -19,7 +19,10 @@ class Profiles {
   static async update(req, res) {
```
Please undo all the changes in this method. Sad that I have to keep talking about this

JuwonAbiola replied:
The changes occurred due to merge conflicts

dinobi commented on server/controllers/Profiles.js:
```diff
@@ -68,6 +68,34 @@ class Profiles {
+        });
+      }
+      delete userProfile.dataValues.password;
+      return serverResponse(res, 200, {
```
can we create userResponse helper thats basically works like serverResponse, but have an added responsibility of deleting the password from the user project before sending to client?

dinobi commented on server/database/migrations/20190813101408-create-role.js:
```diff
@@ -0,0 +1,27 @@
+export default {
```
Why do we have a this in this PR?

dinobi commented on server/controllers/Profiles.js:
```diff
@@ -68,6 +73,29 @@ class Profiles {
+      if (!userProfile) {
+        return serverResponse(res, 404, { error: 'user does not exist' });
+      }
+      return userResponse(res, 200, userProfile);
```
Please use this in other places where we delete password as well.

JuwonAbiola committed:
feature-[167783727]:view user profile
- write controller method to view user profile
- write integration test
[Delivers #167783727]
